### ISSUE TITLE
Empty citations in /messages API response for chatflow app when citations feature is enabled

### ISSUE THREAD

**Issue by hwk68** [langgenius/dify#6368]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.14

### Cloud or Self Hosted

Cloud, Self Hosted (Docker)

### Steps to reproduce

1. Create a chatflow app
2. Use the Knowledge Retrieval node in the chatflow
3. Enable the Citations and Attributions feature
4. Make an API request to the /chat-messages endpoint to start a conversation, ensuring the responses contain citations(retriever_resources[])
5. Make an API request to the /messages endpoint to retrieve the messages from the conversation in the previous step
6. Observe that the retriever_resources[] in the response is empty

### ✔️ Expected Behavior

When using the /messages API endpoint for a chatflow app with the citations feature enabled in the backend, the json response should include relevant citation data in the retriever_resources[] array.

### ❌ Actual Behavior

When the citations feature is enabled, calling the /messages endpoint for a chatflow app results in an empty retriever_resources[] array in the json response.
This issue only occurs with the chatflow(beta) API(Advanced Chat App API); the chatbot API(Chat App API) does not exhibit this problem.

screenshot:

- The response from the /chat-messages endpoint:
![chat-messages](https://github.com/user-attachments/assets/a54d22e0-2411-40bb-8932-d8b83b0c73d2)

- The response from the /messages endpoint:
![messages](https://github.com/user-attachments/assets/4510755b-02ac-4659-aad2-aa48be6f331e)

**Comment by crazywoola** [comment]
I guess that endpoint is missing the reference to those retrieved data.

**Comment by nazeco** [comment]
I have encountered the same issue.

When using the basic orchestration in a chatbot, the knowledge base references can be correctly found in both the streaming chat and the historical records.

However, when switching to chatflow orchestration, they can only be found in the `knowledge-retrieval` node of the streaming chat.

I checked the response of the request history interface, and it also returned an empty `retriever_resources[]` array.

This means that the knowledge base references are only displayed during the chat session and are lost from the chat history after refreshing the page.
